### PR TITLE
Add device templates for Cortex-A53 and Cortex-R4

### PR DESCRIPTION
**Pull request by Masmiseim36**
Hello ARM team,

I have tried to integrate support for Cortex-R and newer Cortex-A types on two core types as an example.

Is this the correct way to do?

### CODE REVIEW

File: Device/ARMCA53/Include/ARMCA53.h [hunk 1]
```diff
@@ -0,0 +1,180 @@
+/**************************************************************************//**
+ * @file     ARMCA53.h
+ * @brief    CMSIS-Core(A) Device Header File for Device ARMCA53
+ *
+ * @version  V1.0.0
+ * @date     18. July 2023
```
Comment (JonatanAntoni) [inline Device/ARMCA53/Include/ARMCA53.h]: Should this be the current date?

File: Device/ARMCA53/Include/system_ARMCA53.h [hunk 2]
```diff
@@ -0,0 +1,65 @@
+/******************************************************************************
+ * @file     system_53.h
```
Comment (JonatanAntoni) [inline Device/ARMCA53/Include/system_ARMCA53.h]:
```suggestion
 * @file     system_ARMCA53.h
```

File: Device/ARMCA53/Source/system_ARMCA53.c [hunk 3]
```diff
@@ -0,0 +1,93 @@
+/******************************************************************************
+ * @file     system_53.c
```
Comment (JonatanAntoni) [inline Device/ARMCA53/Source/system_ARMCA53.c]:
```suggestion
 * @file     system_ARMCA53.c
```

**Review by JonatanAntoni — COMMENTED** [review]
Looks promising.

File: Device/ARMCA53/Include/ARMCA53.h (hunk 1 above)
Comment (Masmiseim36) [reply Device/ARMCA53/Include/ARMCA53.h]: should be the date of comit, right? --> fixed

File: Device/ARMCA53/Source/system_ARMCA53.c (hunk 3 above)
Comment (Masmiseim36) [reply Device/ARMCA53/Source/system_ARMCA53.c]: thanks
done

File: Device/ARMCA53/Include/system_ARMCA53.h (hunk 2 above)
Comment (Masmiseim36) [reply Device/ARMCA53/Include/system_ARMCA53.h]: thanks
done

File: Device/ARMCA53/Config/mem_ARMCA53.h [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+ * @file     mem_ARMCA53.h
+ * @brief    Memory base and size definitions (used in scatter file)
+ * @version  V1.0.0
+ * @date     31. March 2024
+ *
+ * @note
+ *
+ ******************************************************************************/
+/*
+ * Copyright (c) 2009-2019 Arm Limited. All rights reserved.
```
Comment (ghost) [inline Device/ARMCA53/Config/mem_ARMCA53.h:11]: Is the year in the copyright header correct?
I think it should be only 'Copyright (c) 2024 Arm Limited.'

Please check also the other copyright headers.

**Review by ghost — CHANGES_REQUESTED** [review]
Please check the date in the copyright headers.
-> Copyright (c) 2009-2019 Arm Limited.
Every year before 2024 could not be right if the file was created in year 2024.

File: Device/ARMCA53/Config/mem_ARMCA53.h (hunk 4 above)
Comment (Masmiseim36) [reply Device/ARMCA53/Config/mem_ARMCA53.h:11]: All copyrights should now be set to 2024

**Comment by Masmiseim36** [comment]
Do you have any idea how I can fix this warning?

M502: RTE Model reports: #error 511:  ARM::Device:Startup : Additional software components required
M504: RTE Model reports: MISSING:     -- require CMSIS:CORE

**Comment by ghost** [comment]
Hello @Masmiseim36,
These warnings will disappear as soon as the new devices are supported in the ARM.CMSIS.pdsc, which is used when calling packchk.exe.

**Review by ghost — APPROVED** [review]
Looks good to me.

**Review by JonatanAntoni — APPROVED** [review]

**Pull request merged**